docs/tasks/access-kubernetes-api/migrate-third-party-resource.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Migrate a ThirdPartyResource to CustomResourceDefinition
+assignees:
+- enisoc
+- deads2k
+---
+
+{% capture overview %}
+This page shows how to migrate data stored in a ThirdPartyResource (TPR) to a CustomResourceDefinition (CRD).
+
+Kubernetes does not automatically migrate existing TPRs because of API changes introduced as part of
+[graduating to beta](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/thirdpartyresources.md)
+under a new name and API group.
+Instead, both TPR and CRD are available and operate independently in Kubernetes 1.7.
+
+The simplest way to migrate is to stop all clients that use a given TPR, then delete the TPR and
+start from scratch with a CRD.
+This page describes an optional process that eases the transition by migrating existing TPR data for
+you **on a best-effort basis**.
+{% endcapture %}
+
+{% capture prerequisites %}
+* Make sure your Kubernetes cluster has a **master version of exactly 1.7.x** (any patch release),
+  as this is the only version that supports both TPR and CRD.
+* If you use a TPR-based custom controller, check with the author of the controller first.
+  Some or all of these steps may be unnecessary if the custom controller handles the migration for
+  you.
+* Be familiar with the concept of [custom resources](/docs/concepts/extensions/custom-resources/),
+  which were known as *third-party resources* until Kubernetes 1.7.
```

Comment:
Is the confusion because we called the general concept "third-party resources" and the API object  "ThirdPartyResource", and we are renaming both at the same time? We tried to make this better going forward by calling the API object CustomResourceDefinition so we could use the term "custom resource" to refer to the general concept, which is a thing you can achieve either with CRD or some other way. (see #45277)